data/lib/src/datasource/authentication_datasource.dart

```diff
@@ -53,30 +54,30 @@ class AuthenticationDataSource {
           PermanentTokenBodyRequest('LinShare-${deviceManager.getPlatformString()}-$deviceUUID'));
       return permanentToken.toToken();
     }).catchError((error) {
-      _throwAuthenticateException(error);
+        _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
+          if (error.response.statusCode == 401) {
+            throw BadCredentials();
+          } else if (error.response.statusCode == 500) {
+            throw MissingRequiredFields();
```

Comment:
is this necessary for this request?

Reply:
No I guess

Reply:
why this still stay here?, please remove it

Reply:
I bet, you delete in wrong place